src/core/server/rendering/bootstrap/bootstrap_renderer.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createHash } from 'crypto';
+import * as UiSharedDeps from '@kbn/ui-shared-deps';
+import { PackageInfo } from '@kbn/config';
+import { UiPlugins } from '../../plugins';
+import { IUiSettingsClient } from '../../ui_settings';
+import { HttpAuth, KibanaRequest } from '../../http';
+import { getPluginsBundlePaths } from './get_plugin_bundle_paths';
+import { BootstrapTemplateInterpolator } from './template_interpolator';
+
+export type BootstrapRendererFactory = (factoryOptions: FactoryOptions) => BootstrapRenderer;
+export type BootstrapRenderer = (options: RenderedOptions) => Promise<RendererResult>;
+
+interface FactoryOptions {
+  serverBasePath: string;
+  packageInfo: PackageInfo;
+  uiPlugins: UiPlugins;
+  auth: HttpAuth;
+}
+
+interface RenderedOptions {
+  request: KibanaRequest;
+  uiSettingsClient: IUiSettingsClient;
+}
+
+interface RendererResult {
+  body: string;
+  etag: string;
+}
+
+export const bootstrapRendererFactory: BootstrapRendererFactory = ({
+  packageInfo,
+  serverBasePath,
+  uiPlugins,
+  auth,
+}) => {
+  const templateInterpolator = new BootstrapTemplateInterpolator();
+
+  const isAuthenticated = (request: KibanaRequest) => {
+    if (!auth.isEnabled()) {
+      return true;
+    }
+    const { status: authStatus } = auth.get(request);
+    // status is 'unknown' when auth is disabled. we just need to not be `unauthenticated` here.
+    return authStatus !== 'unauthenticated';
+  };
+
+  return async ({ uiSettingsClient, request }) => {
+    let darkMode: boolean;
+    let themeVersion: string;
+
+    try {
+      const authenticated = isAuthenticated(request);
+      darkMode = authenticated ? await uiSettingsClient.get('theme:darkMode') : false;
+      themeVersion = authenticated ? await uiSettingsClient.get('theme:version') : 'v7';
+    } catch (e) {
+      // need to be resilient to ES connectivity issues
+      darkMode = false;
+      themeVersion = 'v7';
+    }
```

Comment:
Added this catch block to fire-proof a little, but it wasn't present in legacy. Tell me if we'd rather not have it.

Reply:
ok for me. Probably it makes sense to initialize variables with these defaults?